src/components/ReloadPrompt.vue

```diff
@@ -0,0 +1,33 @@
+<script setup>
+import { useRegisterSW } from "virtual:pwa-register/vue";
+
+const { offlineReady, needRefresh, updateServiceWorker } = useRegisterSW();
+
+const close = async () => {
+    offlineReady.value = false;
+    needRefresh.value = false;
+};
+</script>
+
+<template>
+    <div v-if="offlineReady || needRefresh" class="pwa-toast" role="alert">
+        <div class="message">
+            <span v-if="offlineReady"> App ready to work offline </span>
+            <span v-else> New content available, click on reload button to update. </span>
+        </div>
+        <button v-if="needRefresh" @click="updateServiceWorker()">Reload</button>
+        <button @click="close">Close</button>
+    </div>
```

Comment:
Copied this from the documentation, CSS needs to be updated for various themes, and these strings need to be localized.